org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/servicetemplates/plans/PlansResource.java

```diff
@@ -164,4 +163,48 @@ private Response saveFile(TPlan tPlan, InputStream uploadedInputStream, FormData
     public PlanResource getEntityResource(@PathParam("id") String id) {
         return this.getEntityResourceFromEncodedId(id);
     }
+
+    /**
+     * Endpoint to generate management plans using the container's plan builder. The plan builder
+     * generates all supported management plans (e.g., build plans, termination plans) at once and
+     * uploads it into Winery.
+     *
+     * @param uriInfo the {@link UriInfo} object for this endpoint
+     */
+    @POST
+    @Path("generate")
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces({MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public Response generatePlans(@Context UriInfo uriInfo) {
+        LOGGER.info("Generating plans for Service Template...");
+        // Only if plan builder endpoint is available
+        String planBuilderBaseUrl =
+            Environment.getUrlConfiguration().getContainerApiUrl() + "/containerapi/planbuilder";
+        if (RestUtils.isResourceAvailable(planBuilderBaseUrl)) {
+            // Determine URIs
+            String plansURI = uriInfo.getAbsolutePath().resolve("../plans").toString();
+            String csarURI = uriInfo.getAbsolutePath().resolve("../?csar").toString();
+            // Prepare XML request
+            String request = "<generatePlanForTopology><CSARURL>";
+            request += csarURI;
+            request += "</CSARURL><PLANPOSTURL>";
+            request += plansURI;
+            request += "</PLANPOSTURL></generatePlanForTopology>";
+            // Create client and execute request (handle exceptions generally)
+            WebResource.Builder wr = Client.create()
+                .resource(planBuilderBaseUrl + "/sync")
+                .type(MediaType.APPLICATION_XML);
+            try {
+                wr.post(String.class, request);
+                LOGGER.info("Plans successfully generated");
+            } catch (Exception e) {
+                LOGGER.error("Could not create plans: {}", e.getMessage(), e);
+                return Response.serverError().entity(e.getMessage()).build();
+            }
+        } else {
+            LOGGER.warn("Plan Builder service is not available. No plans were generated.");
+            return Response.serverError().build();
```

Comment:
Can't you put an entity here? You can use the same string as the warning message. This would help users. I assume the body of the response is rendered in the Angular UI.

Reply:
Done